src/function/aggregate/distributive/mode.cpp

```diff
@@ -151,52 +107,72 @@ struct ModeFunctionString {
 	}
 };
 
-AggregateFunction GetModeFunction(PhysicalType type) {
-	switch (type) {
-	case PhysicalType::UINT16:
-		return AggregateFunction::UnaryAggregateDestructor<ModeState<uint16_t>, uint16_t, uint16_t, ModeFunction>(
-		    LogicalType::UTINYINT, LogicalType::UTINYINT);
-	case PhysicalType::UINT32:
-		return AggregateFunction::UnaryAggregateDestructor<ModeState<uint32_t>, uint32_t, uint32_t, ModeFunction>(
-		    LogicalType::UINTEGER, LogicalType::UINTEGER);
-	case PhysicalType::UINT64:
-		return AggregateFunction::UnaryAggregateDestructor<ModeState<uint64_t>, uint64_t, uint64_t, ModeFunction>(
-		    LogicalType::UBIGINT, LogicalType::UBIGINT);
+template <typename INPUT_TYPE, typename KEY_TYPE>
+AggregateFunction GetTypedModeFunction(const LogicalType &type) {
+	return AggregateFunction::UnaryAggregateDestructor<ModeState<KEY_TYPE>, INPUT_TYPE, INPUT_TYPE,
+	                                                   ModeFunction<KEY_TYPE>>(type, type);
+}
+
+AggregateFunction GetModeAggregate(const LogicalType &type) {
+	switch (type.InternalType()) {
+	case PhysicalType::INT8:
+	case PhysicalType::UINT8:
+		return GetTypedModeFunction<uint8_t, uint8_t>(type);
 	case PhysicalType::INT16:
-		return AggregateFunction::UnaryAggregateDestructor<ModeState<int16_t>, int16_t, int16_t, ModeFunction>(
-		    LogicalType::TINYINT, LogicalType::TINYINT);
+	case PhysicalType::UINT16:
```

Comment:
That will work, but I would prefer to use the "correct" physical type for clarity.

Reply:
My main concern was code bloat. The new version actually has the same number of expanded templates as before, but I can add the new expansions and the linker will probably deal with it.